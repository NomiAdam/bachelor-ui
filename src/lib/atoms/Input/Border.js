import styled from 'styled-components';
import PropTypes from 'prop-types';
import {StyledInput} from './Input';
import {StyledTextArea} from './TextAreaInput';
import {StyledSelect} from './Select';
import { basicTheme, lightTheme } from '../../constants/theme';

const StyledBorder = styled.span`
	position: absolute;
	bottom: ${({select, textArea}) => select ? '-8px' :  textArea ? '3px' : '-4px'}
	left: 0;
	height: 2px;
	width: 100%;
	background: ${({background, error}) => error ? basicTheme.red : background || lightTheme.lightBlue};
	transform: scaleX(0);
	transform-origin: 0 0;
	transition: .5s ease all;
	${StyledInput}:focus ~ & {
		transform: scaleX(1);
	}
	${StyledTextArea}:focus ~ & {
		transform: scaleX(1);
	}
	${StyledSelect}:focus ~ & {
		transform: scaleX(1);
	}
`;

StyledBorder.propTypes = {
	/**
	 * String representing color of border, ie. "red", "blue", ....
	 */
	background: PropTypes.string,
	/**
	 * Whether we use border for Input
	 */
	select: PropTypes.bool,
	/**
	 * Whether we use border for TextArea
	 */
	textArea: PropTypes.bool,
    /**
     * Whether we have validation or other error with TextField
     */
    error: PropTypes.bool,
};

export default StyledBorder;
