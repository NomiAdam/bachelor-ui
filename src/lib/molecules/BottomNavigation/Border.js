import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
    findIndex, length, propEq, __, divide, multiply, compose,
} from 'ramda';
import { basicTheme } from '../../constants/theme';

const StyledBorder = styled.div`
  position: absolute;
  height: 2px;
  width: ${ ({ percentage }) => percentage }%;
  background-color: ${ ({ color }) => color };
  top: 0;
  left: ${ ({ leftPosition }) => leftPosition }%;
  transition: 1s all ease;
`;

const multiplyByHundred = multiply(100);
const divideHundred = divide(100);
const findActiveIndex = active => findIndex(propEq('id', active));

const Border = ({ items, active, color }) => {
    const lengthOfItems = length(items);
    const percentageOfBorder = compose(
        multiplyByHundred,
        divide(__, lengthOfItems),
        findActiveIndex(active),
    );
    return (
        <StyledBorder
            leftPosition={ percentageOfBorder(items) }
            percentage={ divideHundred(lengthOfItems) }
            color={ color }
        />
    );
};

Border.propTypes = {
    /**
     * Array of objects is expected, { id: 'String', icon: 'IconNode', label: 'String' }
     */
    items: PropTypes.array,
    /**
     * String definition of active currently active item
     */
    active: PropTypes.string,
    /**
     * String definition of border color
     */
    color: PropTypes.string,
};

Border.defaultProps = {
    items: [],
    // eslint-disable-next-line react/default-props-match-prop-types
    percentage: '0',
    color: basicTheme.purple,
};

export { StyledBorder };
export default Border;
