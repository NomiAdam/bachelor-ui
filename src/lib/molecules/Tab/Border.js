import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { findIndex, length, equals, __, divide, multiply, compose } from 'ramda';
import { basicTheme } from '../../constants/theme';
import { minTablet, maxTablet } from '../../constants/device';

const StyledBorder = styled.div`
  position: absolute;
  background-color: ${({ color }) => color};
  transition: 1s all ease;
  @media ${minTablet} {
    height: 2px;
    width: ${({percentage}) => percentage}%;
    bottom: 0;
    left: ${({leftPosition}) => leftPosition}%;
  }
  @media ${maxTablet} {
    width: 2px;
    height: ${({percentage}) => percentage}%;
    left: 0;
    top: ${({leftPosition}) => leftPosition}%;
  }
`;

const multiplyByHundred = multiply(100);
const divideHundred = divide(100);
const findActiveIndex = (active) => findIndex(equals(active));

const Border = ({ items, active, color }) => {
    const lengthOfItems = length(items);
    const percentageOfBorder = compose(
        multiplyByHundred,
        divide(__, lengthOfItems),
        findActiveIndex(active)
    );
    return (
        <StyledBorder
            leftPosition={percentageOfBorder(items)}
            percentage={divideHundred(lengthOfItems)}
            color={color}
        />
    );
};

Border.propTypes = {
    /**
     * Array of strings is expected, ['String', 'Another String'],
     */
    items: PropTypes.array,
    /**
     * String definition of active currently active item
     */
    active: PropTypes.string,
    /**
     * String definition of border color
     */
    color: PropTypes.string,
};

Border.defaultProps = {
    items: [],
    percentage: '0',
    color: basicTheme.purple,
};

export { StyledBorder };
export default Border;
