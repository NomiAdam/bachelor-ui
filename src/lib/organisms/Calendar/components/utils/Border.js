import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { basicTheme } from '../../../../constants/theme';

const StyledBorder = styled.div`
  width: 100%;
  height: 0;
  position: absolute;
  z-index: 1;
  left: 0;
  top: ${({ position }) => position * 48 }px;
  border-bottom: 1px solid ${basicTheme.grey};
`;

StyledBorder.propTypes = {
    /**
     * Position from top
     */
    position: PropTypes.number,
};

export default StyledBorder;
