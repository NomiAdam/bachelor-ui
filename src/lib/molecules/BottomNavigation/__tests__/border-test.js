import React from 'react';
import Border, { StyledBorder } from '../Border';
import { shallow } from 'enzyme';
import 'jest-styled-components';

const mockActive = 'test';
const mockSecondActive = 'invalid';
const mockItems = [
    {
        icon: null,
        label: 'mock',
        id: 'test',
    },
    {
        icon: null,
        label: 'mock',
        id: 'invalid',
    }
];

describe('Border', () => {
    it('should be defined', () => {
        expect(Border).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Border />)).toMatchSnapshot();
    });
    it('should be of type StyledBorder', () => {
        expect(shallow(<Border />).type()).toBe(StyledBorder);
    });
    it('should have left property set to zero 0% when first item is active', () => {
        const wrapper = shallow(<Border items={mockItems} active={mockActive} />);
        expect(wrapper.find(StyledBorder)).toHaveStyleRule('left', '0%');
    });
    it('should have left property set to zero 50% when second item is active', () => {
        const wrapper = shallow(<Border items={mockItems} active={mockSecondActive} />);
        expect(wrapper.find(StyledBorder)).toHaveStyleRule('left', '50%');
    });
    it('should have width set to 50% when there are two items', () => {
        const wrapper = shallow(<Border items={mockItems} active={mockActive} />);
        expect(wrapper.find(StyledBorder)).toHaveStyleRule('width', '50%');
    });
});
