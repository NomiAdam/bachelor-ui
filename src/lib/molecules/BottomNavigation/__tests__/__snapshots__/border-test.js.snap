// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Border should match the snapshot 1`] = `
<styled.div
  color="#9a0794"
  leftPosition={-Infinity}
  percentage={Infinity}
/>
`;
